File: 15-03-18/python/01-Biased-Chandan.py
n = int(input())
ar = {}
for i in range(0,n):
    ar[i] = int(input())

for i in range(0,n):
    if(ar[i] == 0):
        tmp = i
        while(ar[tmp]==0):
            tmp -= 1
            if(tmp<=0):
                break
        ar[tmp] = 0

ans = 0
for i in range(0,n):
    ans += ar[i]
print(ans)

File: Mar-18/15-03-18/java/01-Biased-Chandan.java
import java.util.*;

class TestClass {
    public static void main(String args[] ) throws Exception {
        Scanner input = new Scanner(System.in);
        int n=input.nextInt();        
        int ar[]=new int[n];
        for(int i=0;i<n;i++)
            ar[i]=input.nextInt();
        for(int i=0;i<n;i++)
        {
            if(ar[i]==0)
            {
                int tmp=i;
                while(ar[tmp]==0)
                {
                    tmp--;
                    if(tmp<=0)
                        break;
                }
                ar[tmp]=0;
            }    
        }
        int sum=0;
        for(int i=0;i<n;i++)
            sum+=ar[i];
        System.out.println(sum);
    }
}
